### PR TITLE
Add migration.enabled config option

### PR DESCRIPTION
**Pull request by ruflin**
The migration.enabled config option is off by default. Enabling it will add all the alias fields which are used for making Beats 7.x data compatible with 6.x data for queries and dashboards.

A user that upgrades can either set `migration.enabled: true` in the config file or run `-E migration.enabled=true` on the command line to also have the migration aliases in the Elasticsearch index template.

Further changes:

* Add system test to check the generated template
* Fix autodiscovery tests. Were broken as probably the removal of the aliases by default did not work here as it relied on apache2.

### CODE REVIEW

File: libbeat/cmd/instance/beat.go [hunk 1]
```diff
@@ -63,7 +63,7 @@ import (
 	svc "github.com/elastic/beats/libbeat/service"
 	"github.com/elastic/beats/libbeat/template"
 	"github.com/elastic/beats/libbeat/version"
-	"github.com/elastic/go-sysinfo"
+	sysinfo "github.com/elastic/go-sysinfo"
```
Comment (ruflin) [inline libbeat/cmd/instance/beat.go]: Seems like my idea / goimports did this automatically. Interesting.

File: libbeat/cmd/instance/beat.go [hunk 2]
```diff
@@ -608,6 +609,11 @@ func (b *Beat) configure(settings Settings) error {
 		return fmt.Errorf("error initializing logging: %v", err)
 	}
 
+	if b.Config.Migration.Enabled() {
+		// TODO: Not very nice way of doing this
+		template.Migration = true
```
Comment (ruflin) [inline libbeat/cmd/instance/beat.go]: This is a pretty ugly hack to not have to pass this var through all stages up the the template alias processor. @urso I assume you don't like that?

**Comment by ruflin** [comment]
Trying to add support to enable / disable aliases for the Kibana index pattern I stumbled over the following issue: https://github.com/elastic/beats/issues/10013 Also opened https://github.com/elastic/kibana/issues/28570 to get better support in Kibana.

**Comment by ruflin** [comment]
This PR will only address migration aliases for the Elasticsearch index template. A follow up PR will be created to also enable / disable it in Kibana index pattern after https://github.com/elastic/beats/pull/10075 is done.

**Comment by ruflin** [comment]
This is ready for review.

File: filebeat/autodiscover/builder/hints/logs_test.go [hunk 3]
```diff
@@ -328,15 +328,15 @@ func TestGenerateHints(t *testing.T) {
...
+						"module":         "apache",
 						"fileset.stdout": "access",
 						"fileset.stderr": "error",
 					},
 				},
 			},
 			len: 1,
 			result: common.MapStr{
-				"module": "apache2",
+				"module": "apache",
```
Comment (jsoriano) [inline filebeat/autodiscover/builder/hints/logs_test.go]: Was it failing? I left it intentionally after adding the option for moved modules.

File: filebeat/module/apache2/module.yml [hunk 4]
```diff
@@ -1 +0,0 @@
-movedTo: apache
```
Comment (jsoriano) [inline filebeat/module/apache2/module.yml]: This was added to don't break existing configurations.

File: libbeat/tests/system/test_migration.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+        exit_code = self.run_beat(
+            extra_args=[
+                "export", "template",
+                "-E", "setup.template.fields=" + os.path.join(self.working_dir, "fields.yml"),
+            ],
+            config="libbeat.yml")
+
+        assert exit_code == 0
+        assert self.log_contains_count('"type": "alias"') == 0
```
Comment (jsoriano) [inline libbeat/tests/system/test_migration.py]: Should we check for an specific name? I guess we can still have fields of type alias no matter the config for migration. Same for other checks here.

File: filebeat/module/apache2/module.yml (hunk 4 above)
Comment (ruflin) [reply filebeat/module/apache2/module.yml]: this was removed accidentially during the rebase. Will readd it. It should also bring the tests back to green in autodiscovery.

File: filebeat/autodiscover/builder/hints/logs_test.go (hunk 3 above)
Comment (ruflin) [reply filebeat/autodiscover/builder/hints/logs_test.go]: yes, but probably explained by the removal of the other file. I will revert this change.

File: libbeat/tests/system/test_migration.py (hunk 5 above)
Comment (ruflin) [reply libbeat/tests/system/test_migration.py]: yes, will change that. This was a quick win as we didn't have any other aliases.

**Pull request merged**